sycl/test-e2e/Matrix/element_wise_all_sizes_impl.hpp

```diff
@@ -0,0 +1,129 @@
+      if (std::is_same_v<RefT, float>) {
+        auto diff = make_fp32(C[i][j]) - ref;
+        assert(std::fabs(static_cast<float>(diff)) <
+               std::numeric_limits<float>::epsilon());
+      } else if (std::is_same_v<RefT, int>) {
+        assert(C[i][j] == ref);
+      }
+    }
+}
+
+template <typename T, typename RefT, size_t M, size_t K, size_t TM, size_t TK>
+void matrix_verify_add(queue q, big_matrix<T, M, K> &A, nd_range<2> &r,
+                       const RefT ref) {
+  buffer<T, 2> bufA(A.get_data(), range<2>(M, K));
+
+  q.submit([&](handler &cgh) {
+     sycl::accessor accA{bufA, cgh, sycl::read_write};
+
+     cgh.parallel_for(r, [=](nd_item<2> spmd_item) [[intel::reqd_sub_group_size(
+                             SG_SZ)]] {
+       const auto global_idx = spmd_item.get_global_id(0);
+       const auto global_idy = spmd_item.get_global_id(1);
+       const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+       const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+       sub_group sg = spmd_item.get_sub_group();
+       joint_matrix<sub_group, T, use::a, TM, TK, layout::row_major> sub_a;
+
+       if constexpr (std::is_same_v<T, bfloat16>) {
+         joint_matrix_fill(sg, sub_a, bfloat16(5.0));
+       } else if constexpr (std::is_same_v<T, int8_t>) {
+         joint_matrix_fill(sg, sub_a, 5);
+       }
+
+       auto wi_slice_a =
+           sycl::ext::intel::experimental::matrix::get_wi_data(sg, sub_a);
+       for (int i = 0; i < wi_slice_a.length(); i++) {
+         if constexpr (std::is_same_v<T, bfloat16>) {
+           wi_slice_a[i] = wi_slice_a[i] + bfloat16(2.0);
+         } else if constexpr (std::is_same_v<T, int8_t>) {
+           wi_slice_a[i] = wi_slice_a[i] + 2;
+         }
+       }
+
+       ext::intel::experimental::matrix::joint_matrix_store(
+           sg, sub_a,
+           accA.get_pointer() + (sg_startx * TM) * K + sg_starty / SG_SZ * TK,
+           K);
+     }); // parallel for
+   }).wait();
+  assert_ops_ref<T, RefT, M, K>(bufA.get_host_access(), ref);
+}
+
+template <typename Ta, size_t tM, size_t tK> void add_ref() {
+
+  static constexpr size_t MATRIX_M = tM * M_MULTIPLIER;
+  static constexpr size_t MATRIX_K = 128;
+  Ta A[MATRIX_M][MATRIX_K];
+
+  if constexpr (std::is_same_v<Ta, bfloat16>) {
+    float D[MATRIX_M][MATRIX_K];
+    big_matrix<float, MATRIX_M, MATRIX_K> MD((float *)&D);
+  } else if constexpr (std::is_same_v<Ta, int8_t>) {
+    int D[MATRIX_M][MATRIX_K];
+    big_matrix<int, MATRIX_M, MATRIX_K> MD((int *)&D);
+  }
+
+  size_t NDRangeM = MATRIX_M / tM;
+  size_t NDRangeK = MATRIX_K / tK;
+  queue q;
+  nd_range<2> r({NDRangeM, NDRangeK * SG_SZ}, {1, 1 * SG_SZ});
+
+  if constexpr (std::is_same_v<Ta, bfloat16>) {
+    big_matrix<bfloat16, MATRIX_M, MATRIX_K> MA((bfloat16 *)&A);
+    matrix_verify_add<bfloat16, float, MATRIX_M, MATRIX_K, tM, tK>(q, MA, r,
+                                                                   7.0);
```

Comment:
ref which is 7 and operands which are 5 and 2 are spread across the code, so it takes some time to figure out where does 7 come from in the end. I suggest making it parameters of the same function and leaving some comment, so in the end it would be something like:
// checks that 2.0 + 5.0 in element-wise op will be equal to reference 7.0
matrix_verify_add<bfloat16, float, MATRIX_M, MATRIX_K, tM, tK>(7.0, bfloat16(2.0), bfloat16(5.0));
or 
matrix_verify_add<bfloat16, float, MATRIX_M, MATRIX_K, tM, tK>(7, 2, 5);
depending on the type condition

Reply:
Done